### PR TITLE
New docker user guide

### PR DESCRIPTION
**Pull request by Wesley-Du**

### CODE REVIEW

File: docs/docs/DockerUserGuide/index.md [hunk 1]
```diff
@@ -1,173 +1,259 @@
...
-```bash
-sudo docker pull intelanalytics/analytics-zoo:0.7.0-bigdl_0.10.0-spark_2.4.3
-```
+# Analytics Zoo Docker User Guide 
 
-If you need nightly build version of Analytics-Zoo, you can use the lastest tag:
-```bash
-sudo docker pull intelanalytics/analytics-zoo:latest
-```
+In order to simply the Analytics Zoo installation and configuration, Analytics Zoo docker images have been built and provided on Docker Hub. These docker images have been pre-built with all the dependencies and readily configured to run a bunch of Analytics Zoo examples out-of-box. The pre-installed packages are listed at the end of this page.
```
Comment (jason-dai) [inline docs/docs/DockerUserGuide/index.md]: I think we should reduce the length of "introduction"; maybe just leave the first paragraph, and remove the next two.

File: docs/docs/DockerUserGuide/index.md [hunk 2]
```diff
@@ -1,173 +1,259 @@
...
+
+
+
+## Launch an Analytics Zoo Docker Container
+
+A Linux user (CentOS or Ubuntu) can pull a docker image and launch the docker container with one command line. 
+
+#### Getting Ready
+
+1. Fresh install OS such as Linux OS (CentOS, Ubuntu etc), MacOS or Windows.
```
Comment (jason-dai) [inline docs/docs/DockerUserGuide/index.md]: Does it support Windows?

File: docs/docs/DockerUserGuide/index.md [hunk 3]
```diff
@@ -1,173 +1,259 @@
...
+* download-analytics-zoo.sh is used for downloading Analytics-Zoo distributions.
+* start-notebook.sh is used for starting the jupyter notebook. You can specify the environment settings and spark settings to start a - specified jupyter notebook.
+* analytics-Zoo-${ANALYTICS_ZOO_VERSION} is the Analytics-Zoo home of Analytics-Zoo distribution.
+* analytics-zoo-SPARK_x.x-x.x.x-dist.zip is the zip file of Analytics-Zoo distribution.
+* spark-${SPARK_VERSION} is the Spark home.
+* analytics-zoo is cloned from https://github.com/intel-analytics/analytics-zoo, contains apps, examples using analytics-zoo.
+
+## Run Analytics Zoo Jupyter Notebook example in a container
+
+This section depends on the previous section [“Launch Analytics Zoo Docker Container”](#Launch-Analytics-Zoo-Docker-Container). After the user launches the Analytics Zoo docker container, the Jupyter Notebook service can be started and Analytics Zoo jupyter examples are available.
```
Comment (jason-dai) [inline docs/docs/DockerUserGuide/index.md]: Analytics Zoo Jupyter examples

File: docs/docs/DockerUserGuide/index.md [hunk 4]
```diff
@@ -1,173 +1,259 @@
...
+#### Launch Jupyter Notebook service from container, and specify a remote Spark cluster
+Specify a remote Spark cluster when launching an Analytics Zoo docker container with this command line: 
+```
 sudo docker run -itd --net=host \
     -e NotebookPort=12345 \
-    -e NotebookToken="1234qwer" \
+    -e NotebookToken="your-boken" \
     -e http_proxy=http://your-proxy-host:your-proxy-port  \
     -e https_proxy=https://your-proxy-host:your-proxy-port  \
+    -e RUNTIME_SPARK_MASTER=spark://your-spark-master-host:your-spark-master-port \
```
Comment (jason-dai) [inline docs/docs/DockerUserGuide/index.md]: 1) We should use YARN as an example
2) Does it work for Python notebook? @hkvision

File: docs/docs/DockerUserGuide/index.md [hunk 5]
```diff
@@ -1,173 +1,259 @@
...
+#### “Make a Copy” from an existing Analytics Zoo Jupyter Notebook example and model after it
+The user can also select an existing Analytics Zoo Jupyter Notebook example approximating to his own use case, then click "Make a Copy" and start tailing it. 
+
+![notebook7](notebook7.jpg)
+
+## Submit Jupyter Notebook example from container to remote Spark cluster
+
+Users can choose to run Analytics Zoo Jupyter Notebook from a container environment but leverage a remote Spark cluster. 
+
+#### Launch Jupyter Notebook service from container, and specify a remote Spark cluster
```
Comment (jason-dai) [inline docs/docs/DockerUserGuide/index.md]: YARN cluster

File: docs/docs/DockerUserGuide/index.md (hunk 1 above)
Comment (Wesley-Du) [reply docs/docs/DockerUserGuide/index.md]: > I think we should reduce the length of "introduction"; maybe just leave the first paragraph, and remove the next two.

Noted.

File: docs/docs/DockerUserGuide/index.md (hunk 2 above)
Comment (Wesley-Du) [reply docs/docs/DockerUserGuide/index.md]: > Does it support Windows?

@glorysdj please confirm Windows again?

File: docs/docs/DockerUserGuide/index.md (hunk 4 above)
Comment (Wesley-Du) [reply docs/docs/DockerUserGuide/index.md]: > 1. We should use YARN as an example
Confirmed with @glorysdj, currently we support Spark cluster standalone mode, need task to support YARN.

Comment (jason-dai) [reply docs/docs/DockerUserGuide/index.md]: I don't think we can directly run python code on Spark standalone @hkvision ?

Comment (hkvision) [reply docs/docs/DockerUserGuide/index.md]: I suppose currently standalone can only be run using spark-submit.

File: docs/docs/DockerUserGuide/index.md (hunk 2 above)
Comment (glorysdj) [reply docs/docs/DockerUserGuide/index.md]: yes, windows can run Docker now.

File: docs/docs/DockerUserGuide/index.md (hunk 4 above)
Comment (glorysdj) [reply docs/docs/DockerUserGuide/index.md]: we will try run the Docker with remote YARN.

Comment (Wesley-Du) [reply docs/docs/DockerUserGuide/index.md]: This section will be updated when YARN support is confirmed.

Comment (jason-dai) [reply docs/docs/DockerUserGuide/index.md]: > I suppose currently standalone can only be run using spark-submit.

I don't think you can run python for standalone using spark-submit unless all the python lib are installed on each node @jenniew

Comment (hkvision) [reply docs/docs/DockerUserGuide/index.md]: yeah, only when all the dependencies are installed on every node.

Comment (jenniew) [reply docs/docs/DockerUserGuide/index.md]: If all the python dependencies are installed on each node in spark cluster,  launch the docker container by specifying spark master address would run Jupyter notebook on spark cluster, no need spark-submit.

Comment (jason-dai) [reply docs/docs/DockerUserGuide/index.md]: I think we should remove the remote spark cluster for now.

Comment (Wesley-Du) [reply docs/docs/DockerUserGuide/index.md]: > I think we should remove the remote spark cluster for now.

Noted.

File: docs/docs/DockerUserGuide/index.md [hunk 6]
```diff
@@ -1,173 +1,214 @@
...
+In order to simply the Analytics Zoo installation and configuration, Analytics Zoo docker images have been built and provided on Docker Hub. These docker images have been pre-built with all the dependencies and readily configured to run a bunch of Analytics Zoo examples out-of-box on Linux (such as Ubuntu, CentOS), MacOS or Windows. The pre-installed packages are listed at the end of this page. This document provides step-by-step instructions for users to easily start using the Analytics Zoo docker:
 
-### How to use the image
-**To start a notebook directly with a specified port(e.g. 12345). You can view the notebook on http://[host-ip]:12345**
-```bash
-sudo docker run -it --rm -p 12345:12345 \
-    -e NotebookPort=12345 \
-    -e NotebookToken="your-token" \
-    intelanalytics/analytics-zoo:default
+* [Launch an Analytics Zoo Docker container](#Launch-an-Analytics-Zoo-Docker-container)
```
Comment (jason-dai) [inline docs/docs/DockerUserGuide/index.md]: it seems that the link will become small letter after it is converted to html; e.g., see https://github.com/intel-analytics/analytics-zoo/issues/2034

Comment (Wesley-Du) [reply docs/docs/DockerUserGuide/index.md]: changed the internal tags to lower case; will fix 2034 with a new PR.

File: docs/docs/DockerUserGuide/index.md (hunk 1 above)
Comment (Wesley-Du) [reply docs/docs/DockerUserGuide/index.md]: done

File: docs/docs/DockerUserGuide/index.md (hunk 3 above)
Comment (Wesley-Du) [reply docs/docs/DockerUserGuide/index.md]: done

File: docs/docs/DockerUserGuide/index.md (hunk 4 above)
Comment (Wesley-Du) [reply docs/docs/DockerUserGuide/index.md]: spark cluster section removed

File: docs/docs/DockerUserGuide/index.md [hunk 7]
```diff
@@ -1,173 +1,214 @@
...
-```bash
-sudo docker pull intelanalytics/analytics-zoo:0.7.0-bigdl_0.10.0-spark_2.4.3
-```
+# Analytics Zoo Docker User Guide 
 
-If you need nightly build version of Analytics-Zoo, you can use the lastest tag:
-```bash
-sudo docker pull intelanalytics/analytics-zoo:latest
-```
+In order to simply the Analytics Zoo installation and configuration, Analytics Zoo docker images have been built and provided on Docker Hub. These docker images have been pre-built with all the dependencies and readily configured to run a bunch of Analytics Zoo examples out-of-box on Linux (such as Ubuntu, CentOS), MacOS or Windows. The pre-installed packages are listed at the end of this page. This document provides step-by-step instructions for users to easily start using the Analytics Zoo docker:
```
Comment (jason-dai) [inline docs/docs/DockerUserGuide/index.md]: Add a link to the docker hub address for Analytics Zoo

File: docs/docs/DockerUserGuide/index.md [hunk 8]
```diff
@@ -1,173 +1,214 @@
...
+![notebook6](notebook6.jpg)
+
+#### “Make a Copy” from an existing Analytics Zoo Jupyter Notebook example and model after it
+The user can also select an existing Analytics Zoo Jupyter Notebook example approximating to his own use case, then click "Make a Copy" and start tailing it. 
+
+![notebook7](notebook7.jpg)
+
+
+## Terminate the Analytics Zoo Docker container
+Users should terminate the Analytics Zoo Docker container done using it.
```
Comment (jason-dai) [inline docs/docs/DockerUserGuide/index.md]: Fix this sentence

File: docs/docs/DockerUserGuide/index.md (hunk 7 above)
Comment (Wesley-Du) [reply docs/docs/DockerUserGuide/index.md]: Added.

File: docs/docs/DockerUserGuide/index.md (hunk 8 above)
Comment (Wesley-Du) [reply docs/docs/DockerUserGuide/index.md]: changed "done" to "after".

File: docs/docs/DockerUserGuide/index.md [hunk 9]
```diff
@@ -1,173 +1,214 @@
...
-```bash
-sudo docker pull intelanalytics/analytics-zoo:0.7.0-bigdl_0.10.0-spark_2.4.3
-```
+# Analytics Zoo Docker User Guide 
 
-If you need nightly build version of Analytics-Zoo, you can use the lastest tag:
-```bash
-sudo docker pull intelanalytics/analytics-zoo:latest
-```
+In order to simply the Analytics Zoo installation and configuration, Analytics Zoo docker images have been built and provided on [Docker Hub](!https://hub.docker.com/r/intelanalytics/analytics-zoo). These docker images have been pre-built with all the dependencies and readily configured to run a bunch of Analytics Zoo examples out-of-box on Linux (such as Ubuntu, CentOS), MacOS or Windows. The pre-installed packages are listed at the end of this page. This document provides step-by-step instructions for users to easily start using the Analytics Zoo docker:
```
Comment (jason-dai) [inline docs/docs/DockerUserGuide/index.md]: why "!https://hub.docker.com/r/intelanalytics/analytics-zoo"?

Comment (Wesley-Du) [reply docs/docs/DockerUserGuide/index.md]: Per review comment "Add a link to the docker hub address for Analytics Zoo",  this link is added as it is the docker hub home page for Analytics Zoo.

Comment (jason-dai) [reply docs/docs/DockerUserGuide/index.md]: but why "!"? it makes link broken on the md file

Comment (Wesley-Du) [reply docs/docs/DockerUserGuide/index.md]: sorry, fixed (two same mistakes)

File: docs/docs/DockerUserGuide/index.md [hunk 10]
```diff
@@ -1,173 +1,214 @@
...
+
+![notebook6](notebook6.jpg)
+
+#### “Make a Copy” from an existing Analytics Zoo Jupyter Notebook example and model after it
+The user can also select an existing Analytics Zoo Jupyter Notebook example approximating to his own use case, then click "Make a Copy" and start tailing it. 
+
+![notebook7](notebook7.jpg)
+
+
+## Terminate the Analytics Zoo Docker container
```
Comment (jason-dai) [inline docs/docs/DockerUserGuide/index.md]: Use "shut down" instead of "terminate"

Comment (Wesley-Du) [reply docs/docs/DockerUserGuide/index.md]: changed.

**Comment by jason-dai** [comment]
LGTM

**Pull request merged**